### PR TITLE
Refactor code into module that does not depend on zookeeper

### PR DESCRIPTION
**Pull request by hmusum**

### CODE REVIEW

File: zookeeper-server/zookeeper-server-3.5.6/src/main/java/com/yahoo/vespa/zookeeper/LoggingWatcher.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.zookeeper;
+
+import org.apache.zookeeper.WatchedEvent;
+import org.apache.zookeeper.Watcher;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+class LoggingWatcher implements Watcher {
```
Comment (mpolden) [inline zookeeper-server/zookeeper-server-3.5.6/src/main/java/com/yahoo/vespa/zookeeper/LoggingWatcher.java:10]: Consider inlining this as `(event) -> log.log(Level.INFO, event.toString())` so we don't need to have this class in every versioned module.

File: zookeeper-server/zookeeper-server-common/src/main/java/com/yahoo/vespa/zookeeper/ZkAdmin.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+package com.yahoo.vespa.zookeeper;
+
+import java.time.Duration;
+
+/**
+ * Interface for administering a zookeeper cluster. Currently only supports reconfiguring a zookeeper cluster.
+ *
+ * @author hmusum
+ */
+public interface ZkAdmin {
```
Comment (mpolden) [inline zookeeper-server/zookeeper-server-common/src/main/java/com/yahoo/vespa/zookeeper/ZkAdmin.java:11]: Consider naming it `VespaZooKeeperAdmin`, similar to `VespaZooKeeperServer`.

File: zookeeper-server/zookeeper-server-3.5.6/src/main/java/com/yahoo/vespa/zookeeper/LoggingWatcher.java (hunk 1 above)
Comment (hmusum) [reply zookeeper-server/zookeeper-server-3.5.6/src/main/java/com/yahoo/vespa/zookeeper/LoggingWatcher.java:10]: Yes, good point, fixed.

File: zookeeper-server/zookeeper-server-common/src/main/java/com/yahoo/vespa/zookeeper/ZkAdmin.java (hunk 2 above)
Comment (hmusum) [reply zookeeper-server/zookeeper-server-common/src/main/java/com/yahoo/vespa/zookeeper/ZkAdmin.java:11]: Done

**Review by mpolden — APPROVED** [review]
Great!

**Comment by mpolden** [comment]
Test failure in unrelated (RpcServerTest.testRebinding failed).

**Pull request merged**